tetris.py

```diff
@@ -235,15 +248,23 @@ def full_line_clear(self):
         """
         Clears the line once it's fully populated with blocks
         """
-        y_coordinates = [y for (x, y) in self.landed_blocks]
+
+    def block_color(self):
+        color_list = [RED, BLUE, GREEN, YELLOW, PURPLE, ORANGE, PINK, TEAL]
+        color_chart = {shape: color for shape, color in zip(shape_names, color_list)}
+        return color_chart[self.current_block_shape]
+
+    def full_line_clear(self):
+        """
+        Clears the line once it's fully populated with blocks
+        """
+        y_coordinates = [y for (x, y) in self.coord_extractor()]
         coordinates_counter = collections.Counter(y_coordinates)
-        for y_line in range(game_height):
-            count = coordinates_counter[y_line]
+        for x_line in range(game_height):
+            count = coordinates_counter[x_line]
```

Comment:
`x_line` is a terrible variable name, because the value of this variable is an `y` coordinate.